### PR TITLE
Add api_path to connect to rerouted pulp

### LINKED ISSUE #453: As a user, I can have an option to interact with a rerooted Pulp
## Summary
The ability to reroot Pulp is tracked here. https://github.com/pulp/pulpcore/issues/2148

It would be great if a setting could be added that would apply to all commands when set to use a different URL instead of `/pulp/api/v3`.

### PR DESCRIPTION
**Pull request by mdellweg**
fixes #453

### CODE REVIEW

**Comment by mdellweg** [comment]
The tests still use hardcoded paths. I am wondering if we can start the container with a rerouted api...

**Comment by mdellweg** [comment]
The tests still use hardcoded paths. I am wondering if we can start the container with a rerouted api...

File: pulpcore/cli/core/task.py [hunk 1]
```diff
@@ -125,7 +125,7 @@ def cancel(
         try:
             pulp_ctx.wait_for_task(entity)
         except Exception as e:
-            if not re.match("Task /pulp/api/v3/tasks/[-0-9a-f]*/ canceled", str(e)):
+            if not re.match(rf"Task {pulp_ctx.api_path}tasks/[-0-9a-f]*/ canceled", str(e)):
```
Comment (gerrod3) [inline pulpcore/cli/core/task.py:128]: Neat! I did not know you could combine r & f strings.

**Review by gerrod3 — APPROVED** [review]
Looks good, we can add tests in a seprarate PR or wait for @bmbouter's reroute work to be complete.

File: pulpcore/cli/core/task.py (hunk 1 above)
Comment (mdellweg) [reply pulpcore/cli/core/task.py:128]: Yep, i learned that too.

**Comment by mdellweg** [comment]
We'd need a rerouted container to really test this.

**Pull request merged**